src/openrct2/world/Footpath.cpp

```diff
@@ -1279,7 +1279,7 @@ static int32_t footpath_is_connected_to_map_edge_helper(
     // Struct for keeping track of tile state
     struct TileState
     {
-        //bool processed = false;
+        bool processed = false;
         CoordsXYZ footpathPos;
```

Comment:
The description of this third commit states:

1. Fixed a bug that caused the scenario editor to hang when searching for the map edge
2. Also implements vanilla RCT2 behavior when unowning paths by bypassing the early returns when the UNOWN flag is set and letting the method continue to unown all connected path tiles

These are two changes right? So they should go into different commits. It might sound like nitpicking, but imagine that this commits introduce a bug, we won't be able to tell if it was introduced by 1 or 2. It also makes it a little bit harder to review